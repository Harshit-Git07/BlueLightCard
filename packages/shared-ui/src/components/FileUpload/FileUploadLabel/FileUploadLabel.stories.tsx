import { Meta, StoryFn } from '@storybook/react';
import FileUploadLabel from '.';

// Meta data of the component to build the story
const componentMeta: Meta<typeof FileUploadLabel> = {
  title: 'Component System/FileUpload/Label',
  component: FileUploadLabel,
};

// Define the template which uses the component
const DefaultTemplate: StoryFn<typeof FileUploadLabel> = (args) => <FileUploadLabel {...args} />;

export const Default = DefaultTemplate.bind({});
Default.args = {
  labelText: 'label',
};

export default componentMeta;
