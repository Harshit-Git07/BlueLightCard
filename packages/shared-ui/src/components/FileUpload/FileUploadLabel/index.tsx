import { FC } from 'react';
import { colours, fonts } from '../../../tailwind/theme';

export type Props = {
  labelText: string;
};

const FileUploadLabel: FC<Props> = ({ labelText }) => (
  <span className="flex flex-col gap-2">
    <p className={`w-full ${colours.textOnSurface} ${fonts.bodySmallSemiBold}`}>{labelText}</p>
    <p className={`w-full ${colours.textOnSurfaceSubtle}`}>
      Remember to place on plain, well lit surface with no obstructions, blur or glare
    </p>
  </span>
);

export default FileUploadLabel;
