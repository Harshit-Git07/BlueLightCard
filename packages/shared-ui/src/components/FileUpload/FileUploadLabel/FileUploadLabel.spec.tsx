import { render } from '@testing-library/react';
import FileUploadLabel, { Props } from '.';

describe('FileUploadLabel', () => {
  const defaultProps: Props = {
    labelText: 'LABEL',
  };

  // smoke test
  it('should render component without error', () => {
    const { baseElement } = render(<FileUploadLabel {...defaultProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { container } = render(<FileUploadLabel {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
