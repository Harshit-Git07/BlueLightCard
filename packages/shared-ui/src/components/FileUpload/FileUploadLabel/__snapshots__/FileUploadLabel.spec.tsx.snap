// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadLabel matches snapshot 1`] = `
<div>
  <span
    class="flex flex-col gap-2"
  >
    <p
      class="w-full text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body-small-semibold font-typography-body-small-semibold-weight text-typography-body-small-semibold leading-typography-body-small-semibold tracking-typography-body-small-semibold"
    >
      LABEL
    </p>
    <p
      class="w-full text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    >
      Remember to place on plain, well lit surface with no obstructions, blur or glare
    </p>
  </span>
</div>
`;
